src/value_generators/test/DocumentGeneratorTestCases.yml

```diff
@@ -594,6 +594,55 @@ Tests:
     ThenReturns:
     - { "date" : { "$date" : { "$numberLong" : "1369875424299" } } }
 
+  # Date: Random Dates from RandomInt.
+  - Name: RandomWithInt
+    GivenTemplate:
+      date: {^RandomDate: {min: {^RandomInt: {min: 1369875424299, max: 1369875424299}}, max: {^RandomInt: {min: 1369875424300, max: 1369875424300}}}}
```

Comment:
I missed it, do we check that max > min? With generators it will be easy to construct a case in which that is not true.

Reply:
> it's still easy to construct a case where it aribrarily fails depending on the inputs.

This seems a bit...fraught. Would it make sense for there to be a "min" and "max" member-function on the date generators that returns the min/max values that the generator could ever produce and then apply the min/max checking logic at construction time? This is obviously rather tricky with `^Choose` and perhaps others but maybe that's a good thing - meaning maybe it shouldn't be possible to internally use something other than `DateGen` (isA `Generator<boost::time_point>` or whatever the boost type is)?